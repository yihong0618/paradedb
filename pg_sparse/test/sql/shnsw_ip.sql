SET enable_seqscan = off;

CREATE TABLE t (val svector(3));
INSERT INTO t (val) VALUES ('[0,0,0]'), ('[1,2,3]'), ('[1,1,1]'), (NULL);
CREATE INDEX ON t USING shnsw (val svector_ip_ops);

INSERT INTO t (val) VALUES ('[1,2,4]');

SELECT * FROM t ORDER BY val <#> '[3,3,3]';
SELECT COUNT(*) FROM (SELECT * FROM t ORDER BY val <#> (SELECT NULL::svector)) t2;

DROP TABLE t;

CREATE TABLE t (val svector(4));
INSERT INTO t (val) VALUES ('[0,0,0,1]'), ('[3,4,0,2]'), ('[0,2,0,1]'), ('[0,4,0,0]');
CREATE INDEX ON t USING shnsw (val svector_ip_ops);
SELECT * FROM t ORDER BY val <#> '[3,3,0,3]';

DROP TABLE t;
