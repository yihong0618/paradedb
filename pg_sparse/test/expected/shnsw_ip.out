SET enable_seqscan = off;
CREATE TABLE t (val svector(3));
INSERT INTO t (val) VALUES ('[0,0,0]'), ('[1,2,3]'), ('[1,1,1]'), (NULL);
CREATE INDEX ON t USING shnsw (val svectorip_ops);
INSERT INTO t (val) VALUES ('[1,2,4]');
SELECT * FROM t ORDER BY val <#> '[3,3,3]';
   val   
---------
 [1,2,4]
 [1,1,1]
 [1,2,3]
 [0,0,0]
(4 rows)

SELECT COUNT(*) FROM (SELECT * FROM t ORDER BY val <#> (SELECT NULL::svector)) t2;
ERROR:  different svector dimensions 0 and 3
DROP TABLE t;
